### ISSUE TITLE
Documentation 

### ISSUE THREAD

**Issue by pedromorgan** [htacg/tidy-html5#171]
Why am I here.. cos geoffmcl is a mate.. so checking out project....

Can I help with the documentation..

I am familiar with doxygen and indeed create sites for projects..

- eg a standalone site like tidy5..
http://fgx.freeflightsim.org/

or even a doxygen of doxygens..
http://api-docs.freeflightsim.org/

are some good.. public facing ones.. in spare time (freelance hacker)

The brief... 
=================
Well one thing I dont like is having docs outside of the project.. eg on a website with no control and incorrect.

This means that developers can either put corrections in pages in the code.. c code..
or make ammendments and changes.. to the documentation..

Also me need to make the docs sing with the version..
- eg versioning of docs
- autobuilding docs and versininfo..
- link to all relevant projects so one does not go around in circles..
- buildung a website and docs with links to to the project dynamically

and we can autobuild int on a cron with fabfile or alike

**Comment by balthisar** [comment]
The current documentation is at [http://www.html-tidy.org/tidylib_api/](http://www.html-tidy.org/tidylib_api/) and includes the build number when it was built.

Your current open PR offers a _lot_ of additional information, though, and it's definitely a step in the right direction.

Although the generators for making the docs will continue to be in the repo (the shell scripts will be replaced by cmake as the universal build-everything tool), we've decided _not_ to include the generated documentation in the repo. There are several reasons:

- The docs are a build product, and anyone can build them locally.
- We don't distribute binaries of Tidy in the repo.
- It means having to remember to rebuild the docs for every push.
- No ability to accept PR's that affect the docs without regenerating them manually and doing another push.
- The real documentation is the header files.
- In short, they're always _most_ up to date when we don't include them, but provide instructions.

Personally I appreciate good, online documentation as a quick reference rather than switching my editor view to the header. Not everyone agrees with me, but even so we'll continue to support Doxygen. Its generated output is on our website now, in fact.

You bring up a good point about letting it age, though. After 5.0.0 goes official, we'll make sure that the source is updated with every tagged release. Also it's on our gh-pages, and so PR's can be made against it, too.

**Comment by pedromorgan** [comment]
Yes, generated docs should NEVER be in the source..

Am closing this issue as this is more of a forum message, and questions have been asnwered... Ta ;-)
